File: MyBlog.API/DataContracts/Post.cs
using System;
using System.Collections.Generic;

namespace MyBlog.API.DataContracts
{
    public class Post
    {
        public Post(string title, string imageUrl, string content, string previewContent, params string[] tags)
        {
            Title = title;
            ImageUrl = imageUrl;
            Content = content;
            Tags = tags;
            PreviewContent = previewContent;
            DateCreated = DateTime.Now;
            LikeCount = 0;
            DateLoggedInUserLiked = new List<long>();
        }

        public int Id { get; set; }

        public string Title { get; set; }

        public string ImageUrl { get; set; }

        public string Content { get; set; }

        public string[] Tags { get; set; }

        public string PreviewContent { get; set; }

        public DateTime DateCreated { get; set; }

        public int LikeCount { get; set; }

        public List<long> DateLoggedInUserLiked { get; set; } = new List<long>();
    }
}

File: MyBlog-UI/src/app/models/post.ts
export class Post {
    constructor(id: number, imageUrl: string, title: string, previewContent: string, dateCreated: Date, content: string, likeCount: number, tags: string[]) {
        this.id = id;
        this.title = title;
        this.imageUrl = imageUrl;
        this.previewContent = previewContent;
        this.content = content;
        this.dateCreated = dateCreated;
        this.tags = tags;
        this.likeCount = likeCount;
    }

    public likeCount: number;
    public imageUrl: string
    public id: number;
    public title: string;
    public tags: string[];
    public previewContent: string;
    public content: string;
    public dateCreated: Date;
}